server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -155,6 +155,7 @@ private static IndexVersion def(int id, Version luceneVersion) {
      * In branches 8.7-8.11 see server/src/main/java/org/elasticsearch/index/IndexVersion.java for the equivalent definitions.
      */
 
+    @UpdateForV9
     public static final IndexVersion MINIMUM_COMPATIBLE = V_7_0_0;
```

Comment:
By the same logic, why calculate the min compatible TransportVersion? It's only updated on major versions

Reply:
I was using it in TransportVersions because we already have some logic for calculating versions in that class already, and its not much work to extend it for the minimum compatible version. I can just use a hardcoded value for the min transport version if you think its not worth it?

Reply:
I would keep it hardcoded, but I don't feel strongly either way.

Reply:
I'm also not too fussed - I've reverted it